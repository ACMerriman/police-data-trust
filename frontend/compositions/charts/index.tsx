import { Circle, Square } from "./chartSymbols"
import { KeyEntryWrapper } from "./chartKey"
export { Circle, Square, KeyEntryWrapper}
