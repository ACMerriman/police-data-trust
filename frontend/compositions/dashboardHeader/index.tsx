import * as React from "react"
import NPDCLogo from '../../shared-components/logo'
import HamburgerMenu from './assets/Vector.svg'
import DesktopNav from './desktopNav'
import styles from './dashboardHeader.module.css'

export default function DashboardHeader() {
  const {
    wrapper,
    backgroundBanner,
    leftHeader,
    logoContainer,
    titleContainer,
    mobileTitle,
    desktopTitle,
    rightHeader,
    nav,
    buttonContainer,
    mobileNav,
    desktopNav,
  } = styles;

  return (
    <header className={wrapper}>
      <div className={backgroundBanner}>
        <div className={leftHeader}>
          <div className={logoContainer}>
            <NPDCLogo />
          </div>
          <div className={titleContainer}>
            <h2 className={mobileTitle}>N.P.D.C.</h2>
            <h2 className={desktopTitle}>National Police Data Coalition</h2>
            <p>The national index of police incidents</p>
          </div>
        </div>

        <div className={rightHeader}>
          <nav className={nav}>
            <div className={mobileNav}>
              <HamburgerMenu />
            </div>
            <div className={desktopNav}>
              <DesktopNav />
            </div>
          </nav>
          <div className={buttonContainer}>
            <button className="primaryButton" type="button">
              DONATE
            </button>
          </div>
        </div>
      </div>
    </header>
  )
}
