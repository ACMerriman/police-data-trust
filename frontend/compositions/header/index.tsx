import React, { useState } from "react"
import NPDCLogo from './assets/NPDCLogo.svg'
import styles  from './header.module.css'



export default function Header() {
  let [selected, setSelected] = useState('Search');

  const handleNavChange = (e: any) => {
    let value = e.target.innerText;
    setSelected(value);
  }

  const { wrapper, banner, logoTitle, titleContainer, nav, tab, donate, navDonate} = styles;

  return (
    <header className={wrapper}>
      <div className={banner}>
        <div className={logoTitle}>
          <NPDCLogo />
          <div className={titleContainer}>
            <h2>National Police Data Coalition</h2>
            <p>The national index of police incidents</p>
          </div>
        </div>
        <div className={navDonate}>
          <nav>
            <ul className={nav}>
              <li className={selected === 'Search' ? tab : ''} onClick={handleNavChange}>Search</li>
              <li className={selected === 'Profile' ? tab : ''} onClick={handleNavChange}>Profile</li>
              <li className={selected === 'About' ? tab : ''} onClick={handleNavChange}>About</li>
            </ul>
          </nav>
          <button className={donate}type="button">DONATE</button>
        </div>
      </div>
    </header>
  )
}
