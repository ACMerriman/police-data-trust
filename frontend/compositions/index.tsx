import DashboardHeader from "./dashboard-header/dashboard-header"
import EnrollmentCallToAction from './enrollment-call-to-action/enrollment-call-to-action'
import EnrollmentHeader from './enrollment-header/enrollment-header'
import PasswordAid from './password-aid/password-aid'

export { 
  DashboardHeader,
  EnrollmentCallToAction,
  EnrollmentHeader,
  PasswordAid
}
