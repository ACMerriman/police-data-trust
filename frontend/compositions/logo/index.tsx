import React from "react"
import LogoSVG from "./assets/NPDCLogo.svg"

export default function Logo() {
  return (
    <div className="Logo">
      <LogoSVG />
    </div>
  )
}
