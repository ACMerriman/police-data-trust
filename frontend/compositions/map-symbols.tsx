//@ts-nocheck
import * as d3 from "d3"
import { useEffect, useRef, useState } from "react"

export function MarkerLayer({ data, projection }) {
  const ref = useRef()
  const svg = d3.select(ref.current)

  const [circles, setCircles] = useState([])
  const path = d3.geoPath()
  const circleGenerator = d3.geoCircle()

  useEffect(() => {
    if (!data) return

    const populationScale = d3
      .scaleLinear()
      .domain(d3.extent(data.data.features, (d) => d.properties.population))
      .range([1, 50])

    const dataCircs = data.data.features.slice(0, 100).map((d, i) => {
      const circ = circleGenerator({
        r: populationScale(d.properties.population),
        center: d.geometry.coordinates,
      })
      return {
        center: projection(d.geometry.coordinates),
        path: path(circ),
        population: populationScale(d.properties.population),
      }
    })

    setCircles(dataCircs)
  }, [data])

  return (
    <svg ref={ref} viewBox="0, 0, 1200, 700" className="marker-layer">
      {circles.map((c, i) => (
        <circle
          className="circle-marker"
          cx={c.center[0]}
          cy={c.center[1]}
          r={c.population}
          key={`${i}${c.population}`}
          fill="transparent"
          stroke="green"
          strokeWdth="3"
        />
      ))}
    </svg>
  )
}
