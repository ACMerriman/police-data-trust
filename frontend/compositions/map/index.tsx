//@ts-nocheck
import { useEffect, useRef, useState } from "react"
import BaseMap from "./BaseMap"
import useData from "./useData"
import * as d3 from "d3"
import useResizeObserver from "use-resize-observer"
import { MarkerLayer } from "./marker-layer"

export interface Map {}

export default function Map() {
  const data = useData()
  const ref = useRef()

  const [translate, setTranslate] = useState<Coord>([487.5 + 112, 305 + 50])


  // we'll use projection to correctly overlay other geodata on the base map
  // type/scale/translate taken from the baselayer data set
  const projection = d3
    .geoAlbersUsa()
    .scale(1300)
    // .translate([487.5 + 112, 305 + 50])
    .translate(translate)


  return (
    <div
      ref={ref}
      id="map-background"
      style={{
        outline: "1px solid yellow",
        height: "100vh"
      }}>
      <svg id="show-data" viewBox={`0, 0, 1200, 700`}  height="100%" width="100%">
        <MarkerLayer data={data} projection={projection} />
        <BaseMap projection={projection}/>
      </svg>
    </div>
  )
}
