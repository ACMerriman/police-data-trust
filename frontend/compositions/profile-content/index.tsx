import ProfileInfo from './profile-info'
import ProfileType from './profile-type'
import SavedSearches from './saved-searches'
import SavedResults from './saved-results'


export {
  ProfileInfo,
  ProfileType,
  SavedResults,
  SavedSearches
}