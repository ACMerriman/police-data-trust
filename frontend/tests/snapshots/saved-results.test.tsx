import * as React from "react"
import { SavedResults } from "../../compositions"
import { render } from "../test-utils"

it("renders Saved Results page correctly", () => {
  const { container } = render(<SavedResults />)
  expect(container).toMatchSnapshot()
})
