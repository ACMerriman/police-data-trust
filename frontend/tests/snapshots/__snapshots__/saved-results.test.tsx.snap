// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Saved Results page correctly 1`] = `
<div>
  <div
    class="tableWrapper"
  >
    <header
      class="tableHeader"
    >
      <span
        class="tableTitle"
      >
        Saved 
        Results
      </span>
      <button
        class="editButton"
      >
        Edit 
        Results
      </button>
    </header>
    <table
      aria-label="Saved Results"
      class="dataTable"
      role="table"
    >
      <thead
        class="dataHeader"
      >
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            Search Date
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Incident Date(s)
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Incident Type
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Officer(s)
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Subject
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Source
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            View Record
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            Dan Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            Ed Smith, Vince Gilligan
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            Dan Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            John Smith
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            2021/09/07
          </td>
          <td
            role="cell"
          >
            2003/01/01
          </td>
          <td
            role="cell"
          >
            Use of force
          </td>
          <td
            role="cell"
          >
            Ed Smith, Vince Gilligan
          </td>
          <td
            role="cell"
          >
            unknown
          </td>
          <td
            role="cell"
          >
            News Article
          </td>
          <td>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right fa-w-8 expandRecordButton"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
      </tbody>
      <tfoot
        class="dataFooter"
      >
        <tr>
          <td>
            11
             records found
          </td>
          <td
            colspan="4"
          />
          <td>
            Show
             
            <input
              class="dataRowPage"
              max="10"
              min="1"
              type="number"
              value="10"
            />
             
            rows
          </td>
          <td>
             
            <button
              disabled=""
            >
              &lt;
            </button>
             
            1
             of 
            2
             
            <button>
              &gt;
            </button>
             
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
